File: sort revision/minimum_difference.cpp
#include<iostream>
using namespace std;

int difference(int arr[],int n)
{
    sort(arr,arr+n);
    int res = INT16_MAX;

    for(int i=1;i<n;i++)
    {
        res = min(res,(arr[i]-arr[i-1]));
    }

    return res;

    // for (int i = 0; i < n; i++)
    // {
    //     cout<<arr[i]<<" ";
    // }
    
    
    
}

int main()
{
    int arr[] = {10,3,20,12};
    int n = 4;
    cout<<difference(arr,n)<<endl;
    return 0;
}

File: sort revision/minimum_difference.js
let minimum_difference = (arr) => {
  arr.sort((a, b) => a - b);
  let res = arr[1] - arr[0];

  for (let i = 1; i < arr.length - 1; i++) {
    res = Math.min(res, arr[i + 1] - arr[i]);
  }

  console.log(res);
};

let arr = [10,7,1,4];

minimum_difference(arr);